src/sensors/humidity.rs

```diff
@@ -0,0 +1,44 @@
+//! Module providing humidity sensor functionality.
+
+use super::{LocalisedSensorMetadata, SensorMetadata, SensorTemplate, Sensors};
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct HumiditySensor {
+    #[serde(flatten)]
+    pub metadata: LocalisedSensorMetadata,
+    pub unit: String,
+    pub value: f64,
+}
+
+#[derive(Debug, Clone)]
+pub struct HumiditySensorTemplate {
+    pub metadata: SensorMetadata,
```

Comment:
Why not have `LocalisedSensorMetadata` here?

Reply:
I think the idea behind this was to avoid having both `SensorMetadata` and `LocalisedSensorMetadata` in the templates, making the API simpler from the consumers side.
Looking at this again with fresh eyes I do agree it seems like added complexity for little benefit.